climt/_core/constants.py

```diff
@@ -0,0 +1,303 @@
+        """
+
+        for name in constant_descriptions.keys():
+
+            if name in self.current_constants.keys():
+                raise IndexError('{} already present in library.'.format(name))
+            value = constant_descriptions[name]['value']
+            units = constant_descriptions[name]['units']
+            type = constant_descriptions[name]['type']
+
+            if type not in self.constant_categories.keys():
+                raise IndexError('{} is not a valid category.'.format(type))
+
+            self.constant_categories[type] = self.constant_categories[type].union((name,))
+
+            self.current_constants[name] = DataArray(value, attrs={'units': units})
+
+    def modify_constants_in_library(self, constant_descriptions):
+        """
+        Modify existing constants in the library.
+
+        Args:
+
+           constant_descriptions (dict):
+               Dictionary containing the description of the constants.
+               The key should be the name of the constant, and the value
+               should be a dictionary containing the following keys:
+
+               * value (float):
+                   The value assigned.
+               * units (string):
+                   The units of the value, e.g, m/s, J/kg.
+
+        Raises:
+
+            IndexError: If the constant does not exist in the library.
+
+        """
+
+        for name in constant_descriptions.keys():
+
+            if name not in self.current_constants.keys():
+                raise IndexError('{} does not exist in library.'.format(
+                    name))
+            value = constant_descriptions[name]['value']
+            units = constant_descriptions[name]['units']
+
+            self.current_constants[name] = DataArray(value, attrs={'units': units})
+
+
+constant_library = ConstantLibrary()
+
+
+def determine_constant_type(name):
+    """
+    Determine constant type using constant_categories.
+
+    Args:
+
+        name (string):
+            the name of the constant.
+
+    Returns:
+        constant_type (string):
+            the type of the constant.
+    """
+
+    for constant_type in constant_library.constant_categories.keys():
+        if name in constant_library.constant_categories[constant_type]:
+            return constant_type
+
+    raise IndexError("{} not present in any category".format(name))
+
+
+def get_constant(name):
+    """
+    Retrieve constant from the library.
+
+        Args:
+
+            name(string):
+                The name of the constant to be retrieved.
+
+        Returns:
+            constant(DataArray):
+                The constant in a DataArray.
+
+        Raises:
+
+            IndexError: If the constant does not exist in the library.
+    """
+
+    if name not in constant_library.current_constants.keys():
+        raise IndexError("{} not present in the library!".format(name))
+
+    return constant_library.current_constants[name]
+
+
+def remove_constant(name):
```

Comment:
This function should never be necessary. I'd remove it.

Reply:
Agreed.

Reply:
fixed